In [2]:
import pandas as pd

titanic_df = pd.read_csv("train.csv")

## Basic Information

In [4]:
titanic_df.shape

(891, 12)

In [5]:
titanic_df.columns

Index(['PassengerId', 'Survived', 'Pclass', 'Name', 'Sex', 'Age', 'SibSp',
       'Parch', 'Ticket', 'Fare', 'Cabin', 'Embarked'],
      dtype='object')

In [6]:
titanic_df.head()

PassengerId  Survived  Pclass  \
0            1         0       3   
1            2         1       1   
2            3         1       3   
3            4         1       1   
4            5         0       3   

                                                Name     Sex   Age  SibSp  \
0                            Braund, Mr. Owen Harris    male  22.0      1   
1  Cumings, Mrs. John Bradley (Florence Briggs Th...  female  38.0      1   
2                             Heikkinen, Miss. Laina  female  26.0      0   
3       Futrelle, Mrs. Jacques Heath (Lily May Peel)  female  35.0      1   
4                           Allen, Mr. William Henry    male  35.0      0   

   Parch            Ticket     Fare Cabin Embarked  
0      0         A/5 21171   7.2500   NaN        S  
1      0          PC 17599  71.2833   C85        C  
2      0  STON/O2. 3101282   7.9250   NaN        S  
3      0            113803  53.1000  C123        S  
4      0            373450   8.0500   NaN        S

In [7]:
titanic_df.dtypes

PassengerId      int64
Survived         int64
Pclass           int64
Name            object
Sex             object
Age            float64
SibSp            int64
Parch            int64
Ticket          object
Fare           float64
Cabin           object
Embarked        object
dtype: object

In [8]:
titanic_df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 891 entries, 0 to 890
Data columns (total 12 columns):
 #   Column       Non-Null Count  Dtype  
---  ------       --------------  -----  
 0   PassengerId  891 non-null    int64  
 1   Survived     891 non-null    int64  
 2   Pclass       891 non-null    int64  
 3   Name         891 non-null    object 
 4   Sex          891 non-null    object 
 5   Age          714 non-null    float64
 6   SibSp        891 non-null    int64  
 7   Parch        891 non-null    int64  
 8   Ticket       891 non-null    object 
 9   Fare         891 non-null    float64
 10  Cabin        204 non-null    object 
 11  Embarked     889 non-null    object 
dtypes: float64(2), int64(5), object(5)
memory usage: 83.7+ KB


## Specific Infromation

In [10]:
titanic_df[["Age","Fare"]].mean()

Age     29.699118
Fare    32.204208
dtype: float64

In [11]:
titanic_df[["Age","Fare"]].describe()

Age        Fare
count  714.000000  891.000000
mean    29.699118   32.204208
std     14.526497   49.693429
min      0.420000    0.000000
25%     20.125000    7.910400
50%     28.000000   14.454200
75%     38.000000   31.000000
max     80.000000  512.329200

**Description:**
1. The average age of passengers is around 30 years, which suggests that the majority of passengers are young adults, mostly between 20 to 40 years.

2. The fare distribution is highly skewed with an average of 32, but most passengers paid less than 15, and only a few paid extremely high fares above 500.

3. The minimum age is 0.42 years, indicating infants on board, and the maximum age is 80 years, showing the presence of elderly passengers.

4. Some fares are 0, which might correspond to crew members or complimentary tickets.



In [13]:
titanic_df.value_counts("Survived")

Survived
0    549
1    342
Name: count, dtype: int64

In [14]:
titanic_df.value_counts("Age")

Age
24.0    30
22.0    27
18.0    26
19.0    25
30.0    25
        ..
53.0     1
66.0     1
70.5     1
74.0     1
80.0     1
Name: count, Length: 88, dtype: int64

In [15]:
titanic_df['Sex'].value_counts()

Sex
male      577
female    314
Name: count, dtype: int64

In [17]:
titanic_df.value_counts(["Sex","Survived"], normalize= True)

Sex     Survived
male    0           0.525253
female  1           0.261504
male    1           0.122334
female  0           0.090909
Name: proportion, dtype: float64

**Description:**

1. The majority of male passengers did not survive, accounting for approximately 52.53% of all passengers, while only 12.23% of males survived.

2. In contrast, female passengers had a higher survival rate, with 26.15% surviving, while only 9.09% did not survive.

3. This clearly shows that females had a much higher chance of survival than males, reflecting the “women and children first” evacuation policy followed during the disaster.

In [19]:
titanic_df[titanic_df["Sex"] == "female"].value_counts("Survived", ascending=True, normalize = True)

Survived
0    0.257962
1    0.742038
Name: proportion, dtype: float64

Nearly one third population of the females survived

In [27]:
titanic_df[titanic_df["Embarked"] == "S"].value_counts("Sex", ascending=True)


Sex
female    203
male      441
Name: count, dtype: int64

The analysis looks at how many people in the Titanic from those who boarded in Southampton, split by gender. It shows that 203 men boarded  and 441 women boarded out of total 644 people

In [28]:
titanic_df[titanic_df["Embarked"] == "S"].value_counts("Survived", ascending=True)


Survived
1    217
0    427
Name: count, dtype: int64

The analysis looks at how many people survived the Titanic from those who boarded in Southampton. It shows that 217 survived and 427 did not, with a total of 644 people.